### PR TITLE
Error Checking, Movement tweaks, API updates, Fixes

### PR DESCRIPTION
**Pull request by Awzaw**
- [x] This PR actually submits something - don't use this system as a messaging service!
- [x] This PR isn't duplicated - you can check if it is by scanning the small list - link is on the navigation bar for this repository.
- [x] This PR includes appropriate markdown for sections - e.g. code blocks for suggested code.
- [x] This PR description and comments in code is understandable - feel free to use your native language to write if you are not comfortable with English.
- [x] This PR has a single branch for itself in your fork - don't just commit to the master branch of your repository!
- [x] This PR has comments written in clear English - please don't write unreadable comments just for your eyes.

### Mostly fixes for NULL level, some API updates, minor changes to movement (checkJump), removed entity $length and adjusted some Entity sizes
...

### master
...

### CODE REVIEW

File: src/revivalpmmp/pureentities/entity/animal/walking/Ocelot.php [hunk 1]
```diff
@@ -47,6 +47,7 @@
 
 
 class Ocelot extends WalkingAnimal implements IntfTameable, IntfCanBreed, IntfCanInteract, IntfCanPanic {
+    use Tameable;
```
Comment (jasonwynn10) [inline src/revivalpmmp/pureentities/entity/animal/walking/Ocelot.php]: wtf is this here for?

File: src/revivalpmmp/pureentities/CustomLogger.php [hunk 2]
```diff
@@ -8,11 +8,21 @@
 use pocketmine\utils\TextFormat;
 use pocketmine\Worker;
 
-class CustomLogger extends MainLogger {
+class CustomLogger extends \AttachableThreadedLogger {
```
Comment (jasonwynn10) [inline src/revivalpmmp/pureentities/CustomLogger.php]: Why not just extend the MainLogger?? It had preexisting functions which don't need remade

File: src/revivalpmmp/pureentities/entity/animal/walking/Ocelot.php (hunk 1 above)
Comment (95CivicSi) [reply src/revivalpmmp/pureentities/entity/animal/walking/Ocelot.php]: Helps reduce duplicate code between Tameable Entities.

Traits
http://php.net/manual/en/language.oop5.traits.php

File: src/revivalpmmp/pureentities/CustomLogger.php (hunk 2 above)
Comment (95CivicSi) [reply src/revivalpmmp/pureentities/CustomLogger.php]: MainLogger only allows a single existance of itself.  Something about how we were creating the CustomLogger as a child of MainLogger was causing issue #210.  This resolves the problem, still allowing for a custom threaded logger, while we can look for other avenues of approach.

**Pull request merged**